tests/test_detectAndMeasure.py

```diff
@@ -161,6 +164,35 @@ def test_detection_xy0(self):
         self.assertTrue(all(output.diaSources['id'] > 1000000000))
         self.assertImagesEqual(subtractedMeasuredExposure.image, difference.image)
 
+    def test_raise_bad_psf(self):
+        """Detection should raise if the PSF width is NaN
+        """
+        # Set up the simulated images
+        noiseLevel = 1.
+        staticSeed = 1
+        fluxLevel = 500
+        rng = np.random.RandomState(333)
+        kwargs = {"seed": staticSeed, "psfSize": 2.4, "fluxLevel": fluxLevel, "x0": 12345, "y0": 67890}
+        science, sources = makeTestImage(noiseLevel=noiseLevel, noiseSeed=6, **kwargs)
+        matchedTemplate, _ = makeTestImage(noiseLevel=noiseLevel/4, noiseSeed=7, **kwargs)
+        difference = science.clone()
+
+        psfShape = difference.getPsf().computeBBox(lsst.geom.Point2D(0, 0)).getDimensions()
+        psfcI = afwImage.ImageD(lsst.geom.Extent2I(psfShape[1], psfShape[0]))
+        psfNew = rng.randn(psfShape[1], psfShape[0])
+        psfcI.array = psfNew
```

Comment:
Is this guaranteed to always make a psf that will fail the tests in the Task? Would setting all the pixels to zero or -1 be just as good, and more reliable? I know you set the seed above, but I don't think that's guaranteed to produce the same results across all architectures, is it?

Reply:
You can't build a kernel with all zeros, and setting to a constant value doesn't actually raise an error. Running several different noise realizations reveals that that sometimes does result in a kernel that has a valid sigma and doesn't raise. I will instead build a fixed kernel I know should raise.